### ISSUE TITLE
Automatically disable error-prone logging checks for gradle plugins

### ISSUE THREAD

**Issue by iamdanfox** [palantir/gradle-baseline#389]
## What happened

We have lots of repos internally where we maintain gradle plugins, but the `Slf4jLogsafeArgs` and `Slf4jConstantLogMessage` checks are pretty redundant as they are intended for server-side or library usage, whereas Gradle plugins are used pretty much exclusively on local machines.

```
apply plugin: 'java-gradle-plugin'

dependencies {
    // ..
}

tasks.withType(JavaCompile) {
    options.errorprone.errorproneArgs += ['-Xep:Slf4jLogsafeArgs:OFF', '-Xep:Slf4jConstantLogMessage:OFF']
}
```

## What should have happened

baseline-error-prone should detect when the `java-gradle-plugin` plugin has been applied and automatically disable these custom logging checks.

**Comment by iamdanfox** [comment]
I still want this feature request

**Comment by iamdanfox** [comment]
Fixed by #504

**Comment by iamdanfox** [comment]
I think we could add `PreconditionsConstantMessage` to the list of checks that are pretty pointless for gradle plugins
